src/runtime/plugin.ts

```diff
@@ -1,23 +1,29 @@
-import { defineNuxtPlugin, useRuntimeConfig } from '#app';
-import { createClient } from '@content-island/api-client';
-import type { ModuleOptions } from '../models';
+import { defineNuxtPlugin } from '#app';
+import type { ApiClient } from '@content-island/api-client';
 
 export default defineNuxtPlugin(() => {
-  const config = useRuntimeConfig();
-  const options = config.contentIsland as ModuleOptions;
-  if (!options?.accessToken) {
-    console.error(
-      'Content Island token is not provided. Please set the' +
-        ' CONTENT_ISLAND_ACCESS_TOKEN environment variable in a `.env` file' +
-        ' in the root of your project OR add it in the' +
-        ' nuxt.config file under the `contentIsland.accessToken` option.'
-    );
-  }
-  const client = createClient(options);
+  const createApiMethod = (endpoint: 'content' | 'contentlist', raw: boolean = false) => {
+    return async (
+      args: Parameters<ApiClient[typeof endpoint extends 'content' ? 'getContent' : 'getContentList']>[0]
+    ) => {
+      const query = raw ? { raw: 'true', ...args } : { raw: 'false', ...args };
+      return await $fetch(`/api/__content_island/${endpoint}`, { query });
+    };
+  };
+
+  const proxyClient = {
+    getContent: createApiMethod('content' as const, false),
```

Comment:
If we wrap the methods as you are proposing, we are losing the Typescript types that the library provides us.

![image](https://github.com/user-attachments/assets/3bdd1dd6-f2c5-4088-9338-8268b82f2fa4)

 Can we create these endpoints but make them as they are defined?

I mean, something like:

```typescript
import type { ApiClient } from '@content-island/api-client';

  const proxyClient: ApiClient = {
    getContent: async query => await $fetch(`/api/__content_island/content`, { query }),
    getRawContent: async query => await $fetch(`/api/__content_island/raw-content`, { query }),
    ...
  };

  return {
    provide: {
      contentIsland: proxyClient,
    },
  };
```